src/uu/uname/src/uname.rs

```diff
@@ -94,34 +94,43 @@ pub fn uumain(args: Vec<String>) -> i32 {
             .short("o")
             .long(OPT_OS)
             .help("print the operating system name."))
-        .get_matches_from(&args);
+        .get_matches_from(args);
 
-    let argc = args.len();
     let uname = return_if_err!(1, PlatformInfo::new());
     let mut output = String::new();
 
-    if matches.is_present(OPT_KERNELNAME) || matches.is_present(OPT_ALL) || argc == 1 {
+    let all = matches.is_present(OPT_ALL);
```

Comment:
As you are doing more than just using an iterator, could you please also document in the commit message about that kind of change

Reply:
This is basically to get the iterator to work, as getting the length of the iterator doesn’t work (without adding `Clone` to the iterator in `uucore`).

I can add a comment though.